### ISSUE TITLE
Investigate Parsing of Function Expression as Function Declaration

### ISSUE THREAD

**Issue by edgemaster** [resource-reasoning/JavaScriptVerification#2]
test262-es6/test/language/statements/variable/12.2.1-3-s.js
test262-es6/test/language/statements/variable/12.2.1-14-s.js
Both fail with errors of the form
```
Fatal error: exception Pulp_Syntax_Utils.PulpInvalid("Expected statement not Function Declaration. Actual /** @codename anonymous21 */\nfunction () \n{ var eval }\n") 
```

But the text of the eval has a function expression (wrapped in `()`), not a declaration.

**Comment by da319** [comment]
Yes, there is a problem with function expression inside group operator ().
Closure compiler removes group operator, so it is not possible to
distinguish between function expression and function declaration in this
case.

Daiva
